### PR TITLE
Add grpc unit test run, expand testing of VectorFactory

### PR DESCRIPTION
**Pull request by jhamon**
## Problem

There is a small number of unit tests that should only get run when grpc dependencies are installed. These were previously omitted from CI by mistake.

## Solution

Add a step to run these grpc steps. Make it conditional on the `use_grpc` test matrix param.

## Type of Change

- [x] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

### CODE REVIEW

**Review by austin-denoble — APPROVED** [review]
Nice, thanks! 🚢

**Comment by ssmith-pc** [comment]
Thanks!

File: .github/workflows/testing-unit.yaml [hunk 1]
```diff
@@ -28,8 +28,11 @@ jobs:
         with:
           include_grpc: '${{ matrix.use_grpc }}'
           include_types: true
-      - name: Run unit tests
+      - name: Run unit tests (REST)
```
Comment (ssmith-pc) [inline .github/workflows/testing-unit.yaml:31]: I think these do cover the REST surface area, but also a lot of the "other" stuff like common utils, etc.

**Review by ssmith-pc — APPROVED** [review]

File: .github/workflows/testing-unit.yaml (hunk 1 above)
Comment (jhamon) [reply .github/workflows/testing-unit.yaml:31]: Yeah. They're really meant to be "everything that doesn't require GRPC dependencies" unit tests. But that doesn't roll off the tongue 🤣

**Pull request merged**